File: back-end/config-service/src/main/java/com/traveltogether/configservice/document/TravelRequest.java
package com.traveltogether.configservice.document;

import java.util.Date;

import com.traveltogether.configservice.model.EGender;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.traveltogether.configservice.model.ETripType;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Data
public class TravelRequest extends BaseDocument<TravelRequest> {

    private EGender genderLooking;

    private ETripType  tripType;

    private String destination;

    private Date departureDate;

    private Date endDate;

    private String description;

    private boolean active;

    private String userIdCreated;
}

File: app/travel-together-fe/src/models/TravelRequest.ts
import { EGender } from "../api/Enum/EGender";
import { ETripType } from "../api/Enum/ETripType";

export class TravelRequest {
  genderLooking?: EGender;
  tripType?: string;
  destination?: string;
  departureDate?: Date;
  endDate?: Date;
  description?: string;
  active?: boolean;

  constructor(
    genderLooking?: EGender,
    tripType?: string,
    destination?: string,
    departureDate?: Date,
    endDate?: Date,
    description?: string,
    active?: boolean
  ) {
    this.genderLooking = genderLooking;
    this.tripType = tripType;
    this.destination = destination;
    this.departureDate = departureDate;
    this.endDate = endDate;
    this.description = description;
    this.active = active;
  }
}
